### PR TITLE
fix: Cleanup and fix Authenticatable concern exposed helpers

### PR DESCRIPTION
**Pull request by dimkl**
Breaking changes:

- Rename `clerk_verified_session_claims` to `clerk_session_claims`
- Drop `clerk_verified_session_token`
- Drop `clerk_session`
- Drop `clerk_reverify_session!` - not implemented

Changes:

- Drop unused fetch of session resource in ProxyV2

### CODE REVIEW

File: lib/clerk/authenticatable.rb [hunk 1]
```diff
@@ -1,97 +1,79 @@
...
 
-    # Makes a request to the Clerk API to verify the session again and return
-    # the Session object. Subsequent calls to this method will return the cached
-    # Session object.
-    #
-    # NOTE: For better performance, you can instead use `#clerk_verified_session_claims`
-    # which already contains the verified claims as retrieved from the session
-    # token.
+    # Returns the verified claims as retrieved from the session token.
     def clerk_session
```
Comment (gkats) [inline lib/clerk/authenticatable.rb:13]: :question: Shouldn't we name this `session_claims`? 

`session` usually refers to the Session resource.

**Review by gkats — APPROVED** [review]
Left one question regarding naming.

Let's make sure we also update the changelog with the breaking change once this goes out of beta (if needed).

File: lib/clerk/authenticatable.rb (hunk 1 above)
Comment (georgepsarakis) [reply lib/clerk/authenticatable.rb:13]: My 2c: I think I agree with `session_claims` as a name, or perhaps use a uniform naming convention with JS & Go SDKs.

**Review by georgepsarakis — APPROVED** [review]

**Review by agis — APPROVED** [review]
LGTM, besides the `clerk_session` name as @gkats suggested.

**Review by gkats — APPROVED** [review]
:rocket:

**Pull request merged**